src/app/core/admin/admin-entity-details/entity-component-select-component/entity-component-select-component.ts

```diff
@@ -0,0 +1,70 @@
+import { DialogCloseComponent } from "#src/app/core/common-components/dialog-close/dialog-close.component";
+import { PanelComponent } from "#src/app/core/entity-details/EntityDetailsConfig";
+import { EntityRelationsService } from "#src/app/core/entity/entity-mapper/entity-relations.service";
+import { Component, Inject } from "@angular/core";
+import {
+  MAT_DIALOG_DATA,
+  MatDialogModule,
+  MatDialogRef,
+} from "@angular/material/dialog";
+import { MatFormFieldModule } from "@angular/material/form-field";
+import { MatSelectModule } from "@angular/material/select";
+
+@Component({
+  selector: "app-entity-component-select-component",
+  imports: [
+    MatDialogModule,
+    MatFormFieldModule,
+    MatSelectModule,
+    DialogCloseComponent,
+  ],
+  templateUrl: "./entity-component-select-component.html",
+  styleUrl: "./entity-component-select-component.scss",
+})
+export class EntityComponentSelectComponent {
+  options: { label: string; value: PanelComponent }[];
+
+  constructor(
+    private entityRelationsService: EntityRelationsService,
+    private dialogRef: MatDialogRef<
+      EntityComponentSelectComponent,
+      PanelComponent
+    >,
+    @Inject(MAT_DIALOG_DATA) public data: { entityType: string },
+  ) {
+    this.options = [
+      {
+        label: $localize`Form Fields Section`,
+        value: {
+          title: $localize`:Default title:New Section`,
+          component: "Form",
+          config: { fieldGroups: [] },
+        },
+      },
+      {
+        label: $localize`Related-Entity Section`,
+        value: {
+          title: $localize`:Default title:New Related Section`,
+          component: "RelatedEntities",
+          config: {},
+        },
+      },
```

Comment:
if no getEntityTypesReferencingType exists, this option should be disabled with a tooltip to the user explaining "Add a field with type that links to this record to enable this section."